Update to use docker harness

andytson-inviqa opened the pull request:
Upgrade notes mention most of the differences.

andytson-inviqa committed:
Update to use docker harness

andytson-inviqa committed:
Test base

andytson-inviqa committed:
Add changes to php sub harnesses

andytson-inviqa committed:
Use attributes.default for attribute files

so harness.yml can override them

andytson-inviqa committed:
Replace webapp static podmonitor with dynamic one

php-fpm-exporter now will be automatically enabled if podMonitoring is enabled, but can be disabled if enabled set to false

andytson-inviqa committed:
Enable lighthouse only in harness.ymls

So it's easier to see what services are enabled

andytson-inviqa committed:
Add claimName template and per storage enabled flags for PVC

andytson-inviqa committed:
Add upgrade notes

andytson-inviqa committed:
Add a note about the docker harness layer

andytson-inviqa committed:
Match the attribute mentions from docker harness notes

kierenevans commented on UPGRADE.md:
```diff
@@ -4,6 +4,43 @@ In addition to the README's [Harness Upgrade Instructions], please note these sp
+harness attributes:
+
+* `app.services[]: *` -> `services.*.enabled: true` - backwards compatibility removed
+* `backend.*.when`, `fronted.*.when` - now are full bash if conditions, so `-f package.json` becomes `"[ -f package.json ]"`, and `"true"` and `"false"` now is semantically valid
```
```suggestion
* `backend.*.when`, `frontend.*.when` - are now full bash if conditions, so `-f package.json` becomes `"[ -f package.json ]"`, and `"true"` and `"false"` now is semantically valid
```

andytson-inviqa replied:
mind boggles with when to use `are now` and `now are`. If you add `we` then even more derivations

kierenevans commented on src/_base/harness.yml:
```diff
@@ -1,6 +1,8 @@
 harness('inviqa/php'):
   description: A docker based development environment for PHP
+  parentLayers:
+    - inviqa/docker:v0.2.0 # will need to be 0.3.0
```
Todo

kierenevans commented on src/_base/harness/config/commands.yml:
```diff
@@ -1,336 +0,0 @@
-    #!bash(workspace:/)|@
-    passthru $COMPOSE_BIN exec -u build console bash
-
-command('composer %'):
```
This got copied to the docker harness but should live here

andytson-inviqa replied:
now lives in commands-php.yml

kierenevans commented on src/_base/harness/config/commands.yml:
```diff
@@ -1,336 +0,0 @@
-    #!bash(workspace:/)|=
-    passthru $COMPOSE_BIN exec -u build console app database:import "$DATABASE_FILE"
-
-command('harness update existing'):
```
Extra steps for composer that got copied to the docker harness. Perhaps a post-update command is needed for harness specific things?

andytson-inviqa replied:
I've put this command in commands-php as is for now but can think about that

andytson-inviqa replied:
https://github.com/inviqa/harness-docker/pull/128

kierenevans commented on src/_base/harness/config/commands-php.yml:
```diff
@@ -0,0 +1,99 @@
+    COMPOSE_PROJECT_NAME: = @('namespace')
+  exec: |
+    #!bash(workspace:/)|@
+    passthru $COMPOSE_BIN exec -u build console app build:frontend
```
Will need code owner here

kierenevans commented on src/_base/harness/config/commands-php.yml:
```diff
@@ -0,0 +1,99 @@
+  exec: |
+    #!bash(workspace:/)|@
+    # Use `bash -i` to load up /home/build/.bashrc, which sets up node version manager (nvm) paths
+    passthru $COMPOSE_BIN exec -u build --workdir '@('frontend.path')' console bash -i
```
Will need code owner here

kierenevans commented on src/_base/harness/config/commands-php.yml:
```diff
@@ -0,0 +1,99 @@
+  exec: |
+    #!bash(workspace:/)|@
+    # Use `bash -i` to load up /home/build/.bashrc, which sets up node version manager (nvm) paths
+    $COMPOSE_BIN exec -u build --workdir '@('frontend.path')' console bash -i -c '@('frontend.watch')'
```
Will need code owner here

kierenevans commented on src/_base/harness/config/commands-php.yml:
```diff
@@ -0,0 +1,99 @@
+  exec: |
+    #!bash(workspace:/)|=
+    echo "Importing Provided Tideways CLI Key (from https://app.tideways.io/user/cli-import-settings)"
+    $COMPOSE_BIN exec -T -u build console tideways import "$TIDEWAYS_SERVERKEY"
```
Will need code owner here

andytson-inviqa committed:
Remove COMPOSE_PROJECT_NAME as is now docker-compose.yml config

andytson-inviqa committed:
Remove docker compose v1 workaround for docker cp

andytson-inviqa committed:
Update tests to use alpha6

andytson-inviqa committed:
Update commands using compose to use CODE_OWNER

andytson-inviqa committed:
Fix upgrade typos

andytson-inviqa committed:
Add php specific commands back in

Interim harness update existing override till possible new command can be embedded in harness-docker to be overriden for post-update tasks

andytson-inviqa committed:
Update tests to use 0.3.0-alpha7 docker harness

and make adjustments for its changes

andytson-inviqa committed:
Remove unneeded raw twig filters

andytson-inviqa committed:
Replace `harness update existing` override with post-update

andytson-inviqa committed:
Update to use docker harness 0.3.0

kierenevans commented on UPGRADE.md:
```diff
@@ -24,6 +94,116 @@ This was in order to remove the use of intermediate image stage builds in favour
+
+### Helm templates pod affinity no longer used for pod spread across zones and hosts
+
+Topology spread constraints are better suited to pod spread than anti-affinity, as will continue to spread more evenly once replicas is higher than the ordinal constraint (e.g. 6 replicas across 3 zones will try to prioritise 2 replicas per zone for).
```
```suggestion
Topology spread constraints are better suited to pod spread than anti-affinity, as will continue to spread more evenly once replicas is higher than the ordinal constraint (e.g. 6 replicas across 3 zones will try to prioritise 2 replicas per zone).
```

kierenevans reviewed (dismissed):
Just an upgrade.md suggestion but can merge. Great job on this.

andytson-inviqa committed:
Apply suggestions from code review

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>

andytson-inviqa committed:
Also change app-migrate to pod template chart.selectors

andytson-inviqa commented:
sorry, I did another pass of the changes for my own review and spotted an inconsistency in app-init/migrate changes
